src/config.rs

```diff
@@ -11,6 +11,18 @@ pub struct Config {
     /// Host port
     pub port: u16,
 
+    #[structopt(
+        long,
+        short = "i",
+        conflicts_with_all(&["host", "port"])
```

Comment:
> * I'm not really happy with the option parsing.  Having just a switch e.g. `static-web-server --inherit-listener` seems reasonable, but it doesn't look like you can do that and also have an ENV variable control the same thing.  I'll have a closer look at this.

Yeah the parsing sometimes is not obvious.
But it's easy to fix just append the `parse(try_from_str), default_value = "false"` to the `#[structopt(...` macro if you want to use boolean options with default values. Then you will be able to use just `-i`.

Reply:
Well, after a quick research turns out it is not possible via `structopt` (based on `clap` args parser). However looks like is doable just using `clap` directly but the downside is we need to redo `src/config.rs` from the scratch which is something that I don't think so happening now.